python/examples/tuning/test_nevergrad_small_matmul_noisy.py

```diff
@@ -0,0 +1,285 @@
+               problem_sizes: Sequence[int]):
+    # TODO: better factoring of these constants
+    self.entry_point_name = 'main'
+    self.fun_to_benchmark_name = 'matmul'
+
+    self.problem_definition = problem_definition
+    self.problem_sizes = problem_sizes
+
+    # Init nevergrad's instrumentation.
+    # TODO: better parametrization, for now this is enough to get a prototype.
+    self.register_tile_sizes = ng.p.Choice(list(range(33)), repetitions=3)
+
+    self.search_keyword = 'search_sizes'
+    self.instrumentation = ng.p.Instrumentation(
+        search_sizes=self.register_tile_sizes)
+    self.optimizer = None
+
+  def set_optimizer(self, search_strategy: str, budget: int):
+
+    def constraints_fun(proposal):
+      return dispatch_size_constraints_conjunction_satisfied( \
+        self.problem_sizes, proposal, self.search_keyword)
+
+    self.search_strategy = search_strategy
+    self.optimizer = ng.optimizers.registry[self.search_strategy](
+        parametrization=self.instrumentation, budget=budget)
+    self.optimizer.parametrization.register_cheap_constraint(constraints_fun)
+
+  # Unwrap the np.array from NG's ask() kwargs.
+  def extract_search_sizes_from_proposal(self, proposal):
+    return [x for x in proposal.kwargs[self.search_keyword]]
+
+  # Optimizer may want to override our constraints set.
+  def validate_proposal(self, proposal):
+    if not size_constraints_conjunction_satisfied(
+        self.problem_sizes, self.extract_search_sizes_from_proposal(proposal)):
+      return False
+    return True
+
+  # TODO: generate a unique name for the matcher.
+  # TODO: generate a tight matche for the generic (e.g. isMatmulOp with
+  # fixed size).
+  def matcher(self, module, benefit: int = 1):
+    pdl_pattern = None
+    pdl_pattern_name = 'foo'
+    with InsertionPoint(module.body):
+      pdl_pattern = pdl.PatternOp(benefit=benefit, name=pdl_pattern_name)
+      with ir.InsertionPoint(pdl_pattern.body):
+        args = pdl.OperandsOp()
+        types = pdl.TypesOp()
+        # TODO: matcher constraints
+        pdl_op = pdl.OperationOp('linalg.generic', args=[args], types=[types])
+        pdl.RewriteOp(pdl_op, 'linalg_transform.apply')
+    return pdl_pattern_name
+
+  # TODO: more advanced schedules, atm we just TileAndVectorize.
+  def schedule(self, module, proposal, benefit: int = 1):
+    search_sizes = self.extract_search_sizes_from_proposal(proposal)
+
+    print(f'Problem sizes: {self.problem_sizes}')
+    print(f'Register tile sizes: {search_sizes}')
+
+    # TODO: this is necessary to force-load the dialect, otherwise op creation
+    # complains about "unregistered dialect" despite the registration called.
+    register_sandbox_passes_and_dialects(module.context)
+    module.context.dialects["linalg_transform"]
+
+    nnz = np.count_nonzero(search_sizes)
+    peel = [x for x in range(nnz)]
+    peel = []  # TODO: enable peeling once handles are fixed.
```

Comment:
Could you elaborate what's the issue with handles that is blocking here?

Reply:
I experienced crashes and did not investigate further, will give it another show once this is refactored.